cmd/erasure-object.go

```diff
@@ -145,23 +145,45 @@ func (er erasureObjects) GetObjectNInfo(ctx context.Context, bucket, object stri
 		return nil, err
 	}
 
+	var nsUnlocker = func() {}
```

Comment:
A slightly 'safer' way would be to:

```suggestion
	var nsUnlocker func()
	defer func() {
		if nsUnlocker != nil {
			nsUnlocker()
		}
	}
```
... and set `nsUnlocker` to `nil` when passing it on. That way you avoid having to call it on every error return path.